src/compiler/transformers/ts.ts

```diff
@@ -3249,9 +3249,10 @@ namespace ts {
 
                 const substitute = createLiteral(constantValue);
                 if (!compilerOptions.removeComments) {
-                    const propertyName = isPropertyAccessExpression(node)
-                        ? declarationNameToString(node.name)
-                        : getTextOfNode(node.argumentExpression);
+                    const originalNode = getOriginalNode(node, isAccessExpression);

```

Comment:
Just curious, what was this one needed for?

Reply:
Template literals and string literals with extended Unicode escapes will be first processed with es2015 transformer which will replace them with `Synthesized` string literals that don't have reference to parent element or to original node.

Under the hood `getTextOfNode(node.argumentExpression)` will call `getSourceFileOfNode` on `Synthesized` string literal which will return `undefined` (because `node.parent` is `undefined`) and then [`getSourceTextOfNodeFromSourceFile`](https://github.com/microsoft/TypeScript/blob/3dd7b84ef82f5985d15b20ac0c150302bcdc49d9/src/compiler/utilities.ts#L522-L524) that will try to access `sourceFile.text` and exception will be thrown.

On the other hand `ElementAccessExpression` even if will be replaced with `Synthesized` node, it will have link to original node. Therefore `originalNode.argumentExpression` will have proper parent references up to source file and text will be retrieved correctly.